Feature/init movement demande de vérification

nyanloutre opened the pull request:
Voila j'ai fait cette fonction partiellement (seulement la touche gauche est gérée)
Du coup je vous envoie ça et vous me dites en commentaire si c'est bien ou si il y a des modifs à faire
Après je rajoute pour les autres touches et on valide le merge

EDIT: Surtout ne validez pas checkez juste la branche chez vous et vérifiez si le comportement est correct

Commit:
Ajout d'un getter sur la liste de tuiles

Commit:
Ajout de deux fonctions pour trier les tuiles

Commit:
Algoritmes de tri de liste de Tiles

Commit:
Tri les tuiles dans des liste différentes pour chaques lignes et tri par coordonées
ou colonnes si on appuie sur up ou down
/!\ cette dernière partie n'est pas finie

Commit:
Recup des positions extrèmes

Commit:
Création de deux liste pour les coordonées possibles

Commit:
Ajout d'un getter pour récupérer le point d'arrivée

Commit:
Gestion de la fusion des tuiles

Commit:
On ne peut pas récupérer le point d'arrivée comme ça sinon on ne peut pas définir la tuile d'arrivée sur null

Commit:
Commentaires pour aider à comprendre

Commit:
On commente cette ligne pour le debug

Commit:
Génération de plus de tuiles pour le debug

Whiax commented:
Du coup le code marche là ? 
Tu penses que ce serait possible de pas refaire des sort à chaque fois ? En mémorisant ton tableau à deux dimensions dans le TileListManager au lieu d'une liste simple

Whiax commented:
Exception in thread "main" java.lang.Error: Unresolved compilation problem: 
    The method sort(TileList.PointCompareY) is undefined for the type List<Tile>

```
at gamepack.TileList.sortY(TileList.java:91)
at gamepack.TileListManager.initMovement(TileListManager.java:109)
at gamepack.WindowGame.keyPressed(WindowGame.java:100)
at org.newdawn.slick.Input.poll(Input.java:1182)
at org.newdawn.slick.GameContainer.updateAndRender(GameContainer.java:641)
at org.newdawn.slick.AppGameContainer.gameLoop(AppGameContainer.java:411)
at org.newdawn.slick.AppGameContainer.start(AppGameContainer.java:321)
at gamepack.WindowGame.main(WindowGame.java:118)
```

Whiax committed:
Collections.sort()

Commit:
Correction classpath

Whiax committed:
On teste pas le refreshFusion si il n'y a pas de tile d'arrivée

Whiax committed:
Pour tester (tile transparente)

Vskilet committed:
Modification de Tile.equals()
Gérer le cas ou secondeTile est null

Commit:
Utilisation var pour stocker emplacement en collone

Whiax committed:
FPS changé

Commit:
Correction X

Commit:
génération de plus de tuiles

Commit:
Correction initMovement

Commit:
Corrections

Commit:
Déplacement de l'inversion de la liste + modif accents

J'ai mis l'inversion de la liste à la fin ca à l'air de marcher chez moi
mais j'ai pas fais beaucoup de tests

Commit:
Bidule qui semble marcher... OH YEAH !

J'ai fais n copier coller brutal et j'ai changé ce qui devait l'être...
et ca semble fonctionner :)

fstaine commented:
Je pense oui, mais je laisse ca aux grands manitous de Git ! 
Par contre chez moi les tuiles se téléportent c'est normal ?

Vskilet committed:
Merge remote-tracking branch 'origin/master' into feature/InitMovement

Conflicts:
	.gitignore
	src/gamepack/TileListManager.java
	src/gamepack/WindowGame.java
